### ISSUE TITLE
p2 organizations extension for tiered organization

### ISSUE THREAD

**Issue by anothergoodguy** [p2-inc/keycloak-orgs#173]
Dear Team,

Great suite of extensions. 

We are working on a retail SaaS usecase that serves at various tiers like:
Tier-1. Retail Corporation
Tier-2. Regional Divisions 
Tier-3. Individual Stores

In this setup, the top tier represents the entire retail corporation, with roles and groups applicable across all regions and stores. 
The middle tier represents individual regional divisions, each with its own set of roles and groups specific to that region. 
The bottom tier represents individual stores, with roles and groups tailored to the needs of each store. 
This tiered organization structure should allow for effective management of authentication and authorization across different levels of the retail business.

I have noticed an entity `Team`; looks like it's from an old code that was not fully implemented as per [keycloak forum](https://keycloak.discourse.group/t/keycloak-multi-tenancy-extensions-for-saas-applications/15426/36) 

Request your help on any guidance / documentation with the phase-two organizations extension. 

Thanks in advance!

**Comment by MGLL** [comment]
Hello, 

Is the topic related to what I would call "Account Hierarchy"? Where an organization can be the child of another organization?
- Note that I have a similar use-case if it's just hierarchy (Country level org > Regional level org) and I'm currently into the thinking process.


Or is it "Tiering" like assigning Subscription tiers to organizations? (through role for example where you could assign a "premium" role to the organization and all users in the organization inherit it)
- Also have a similar use-case, I will work on that soon. 

Best

**Comment by anothergoodguy** [comment]
@MGLL I was thinking on the lines of hierarchy of organizations... as I explained in the [request](https://github.com/p2-inc/keycloak-orgs/issues/173#issue-2131253176).

**Comment by anothergoodguy** [comment]
@MGLL could you share any progress on this please?

**Comment by anothergoodguy** [comment]
hi Team,

Could you please share if there is any progress on this?

Best

**Comment by xgp** [comment]
Hi @anothergoodguy we haven't had any customer interest in this. If you want a feature like this, I'd suggest rouging out a design and proposing it here. But we won't do the work for this unless we have customer need.

**Comment by anothergoodguy** [comment]
thanks, it was much easier to implement with CASBIN, so we took that route

**Comment by pnzrr** [comment]
@anothergoodguy do you leverage a combo of CASBIN and Keycloak for the system's implementation?

**Comment by anothergoodguy** [comment]
yes, keycloak and a separate sidecar micro-service. it was clean, straight forward and much easier to maintain, at the lease for us...

**Comment by xgp** [comment]
Please share details of your implementation, if possible. I’m sure other people could benefit from your insights.

**Comment by anothergoodguy** [comment]
we used it in our proprietary product, will see if I get approval to share...
